### PR TITLE
Make models extensible

### PR DESCRIPTION
**Pull request by pushnitsa**

### CODE REVIEW

File: VirtoCommerce.MarketingModule.Data/Model/PublishingGroupContentPlaceEntity.cs [hunk 1]
```diff
@@ -1,23 +1,49 @@
...
+            }
+
+            pkMap.AddPair(contentPlace, this);
+
+            DynamicContentPlaceId = contentPlace.Id;
+
+            return this;
+        }
+
+        public virtual DynamicContentPlace ToModel(DynamicContentPlace contentPlace)
```
Comment (tatarincev) [inline VirtoCommerce.MarketingModule.Data/Model/PublishingGroupContentPlaceEntity.cs]: I'm not sure what you need a pair FromModel and ToModel for this 'bridge' type

File: VirtoCommerce.MarketingModule.Data/Model/PublishingGroupContentItemEntity.cs [hunk 2]
```diff
@@ -1,22 +1,58 @@
...
         public string DynamicContentItemId { get; set; }
-		public virtual DynamicContentItemEntity ContentItem { get; set; }				
-		#endregion
-	}
+        public virtual DynamicContentItemEntity ContentItem { get; set; }				
+        #endregion
+
+        public int Priority { get; set; }
+
+        public DynamicContentItem ToModel(DynamicContentItem contentItem)
```
Comment (tatarincev) [inline VirtoCommerce.MarketingModule.Data/Model/PublishingGroupContentItemEntity.cs]: I'm not sure what you need a pair FromModel and ToModel for this 'bridge' type

**Comment by pushnitsa** [comment]
> I think there is need some unit tests. At least a test with extended models and its ToCore/ToWeb**** conversions.

In my mind, it doesn't make a sense. We should have a tests for Abstract factory. Why we need a unit tests for extension methods without logic?

File: VirtoCommerce.MarketingModule.Data/Model/PublishingGroupContentItemEntity.cs (hunk 2 above)
Comment (pushnitsa) [reply VirtoCommerce.MarketingModule.Data/Model/PublishingGroupContentItemEntity.cs]: Definitely we need. If we working with content items in group context. For example for prioritization in the groups.

**Pull request merged**